pilot/pkg/serviceregistry/kube/controller/endpointslice.go

```diff
@@ -80,7 +80,7 @@ func newEndpointSliceController(c *Controller) *endpointSliceController {
 
 // TODO use this to automatically switch to EndpointSlice mode
 func endpointSliceV1Available(client kubelib.Client) bool {
-	return client != nil && kubelib.IsAtLeastVersion(client, 21)
+	return client != nil && !kubelib.IsLessThanVersion(client, 21)
```

Comment:
Why is this changed?

Reply:
We should use `EndpointOnly` when getting the version fails, otherwise clusters less than vertion 21 will try to watch endpointslice(v1).